datatypes/src/collections/data_types.rs

```diff
@@ -101,3 +101,47 @@ impl TypedFeatureCollection {
         }
     }
 }
+
+/// Implements a function by forwarding its output
+macro_rules! impl_function_by_forwarding_ref {
```

Comment:
can't we use something like this, instead of delegation in each method?
`fn common(&self> -> Box<dyn &FeatureCollectionOperations>)`
or some Deref magic

Reply:
also: do we even need the FeatureOperations trait if you call it on a concrete type here anyway?

Reply:
We could say `deref()` gives you &dyn FeatureCollectionOperations but it is unclear if this is faster than delegating the functions or even handier when working with either collections directly or typed collections (and refs).

Reply:
The result of the discussion was that it is not possible to do it for operations that return the type itself, so all is done via forwarding to provide a common interface.